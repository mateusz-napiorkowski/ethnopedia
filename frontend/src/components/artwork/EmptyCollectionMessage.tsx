import { HiOutlineCollection } from "react-icons/hi";
import { useNavigate, useParams } from "react-router-dom";

type Props = {
    setShowImportOptions: (value: boolean) => void;
};

const EmptyCollectionMessage = ({ setShowImportOptions }: Props) => {
    const navigate = useNavigate();
    const { collection } = useParams();

    return (
        <div className="px-4 max-w-screen-xl pt-10 pb-10 py-4 bg-white dark:bg-gray-800 shadow-md w-full rounded-lg mb-4 border border-gray-300 dark:border-gray-600 cursor-pointer text-center">
            <HiOutlineCollection className="mx-auto w-16 h-16 mb-4 text-gray-400" />
            <p className="text-xl mb-4">Ta kolekcja jest pusta.</p>
            <p className="text-md">
                <button
                    type="button"
                    className="text-blue-600 cursor-pointer bg-transparent border-0 p-0"
                    onClick={() =>
                        navigate(`/collections/${collection}/create-artwork`)
                    }
                >
                    Dodawaj nowe rekordy
                </button>{" "}
                ręcznie lub{" "}
                <button
                    type="button"
                    className="text-blue-600 cursor-pointer bg-transparent border-0 p-0"
                    onClick={() => setShowImportOptions(true)}
                >
                    zaimportuj
                </button>{" "}
                dane z pliku Excel, aby rozpocząć organizację swojej kolekcji.
            </p>
        </div>
    );
};

export default EmptyCollectionMessage;
