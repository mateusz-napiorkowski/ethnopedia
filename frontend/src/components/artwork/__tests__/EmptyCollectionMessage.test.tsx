import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import EmptyCollectionMessage from "../EmptyCollectionMessage"
import { MemoryRouter, Routes, Route } from "react-router-dom";
import userEvent from "@testing-library/user-event";

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
}));
const mockSetShowImportOptions = jest.fn()

const jwtToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3Rvd3kiLCJmaXJzdE5hbWUiOiJ0ZXN0b3d5IiwidXN"
    + "lcklkIjoiNjZiNjUwNmZiYjY0ZGYxNjVlOGE5Y2U2IiwiaWF0IjoxNzI0MTg0MTE0LCJleHAiOjE3MjUxODQxMTR9.fzHPaXFMzQTVUf9IdZ0G6oeiaecc"
    + "N-rDSjRS3kApqlA"
const collectionId = "6819ce5713049d642d1e72c9"

const user = userEvent.setup()


const renderComponent = (jwtToken: string | undefined = undefined) => {
    return render(
        <MemoryRouter initialEntries={[`/collections/${collectionId}/artworks`]}>
            <Routes>
                <Route path="/collections/:collectionId/artworks" element={<EmptyCollectionMessage setShowImportOptions={mockSetShowImportOptions} jwtToken={jwtToken}/>}/>
            </Routes>  
        </MemoryRouter>
    );
};

describe("EmptyCollectionMessage", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });
    
    it("should render EmptyCollectionMessage component with message for not logged in user", () => {
        const {container} = renderComponent()
        expect(container).toMatchSnapshot()
    })

    it("should call useNavigate(/login) after login button is clicked", async () => {
        const {getByText} = renderComponent()
        const loginButton = getByText(/zaloguj się/i)

        await user.click(loginButton)

        expect(mockUseNavigate).toHaveBeenCalledWith("/login")
    })

    it("should call useNavigate(/register) after register button is clicked", async () => {
        const {getByText} = renderComponent()
        const registerButton = getByText(/zarejestruj się/i)

        await user.click(registerButton)
        
        expect(mockUseNavigate).toHaveBeenCalledWith("/register")
    })

    it("should render EmptyCollectionMessage component with message for logged in user", () => {
        const {container} = renderComponent(jwtToken)
        expect(container).toMatchSnapshot()
    })

    it("should call useNavigate(/collections/:collectionId/create-artwork) after add new record button is clicked", async () => {
        const {getByText} = renderComponent(jwtToken)
        const addNewRecordButton = getByText(/dodawaj nowe rekordy/i)

        await user.click(addNewRecordButton)

        expect(mockUseNavigate).toHaveBeenCalledWith(`/collections/${collectionId}/create-artwork`)
    })

    it("should call setShowImportOptions(true) after add new record button is clicked", async () => {
        const {getByText} = renderComponent(jwtToken)
        const importRecordsButton = getByText(/zaimportuj/i)

        await user.click(importRecordsButton)

        expect(mockSetShowImportOptions).toHaveBeenCalledWith(true)
    })
})