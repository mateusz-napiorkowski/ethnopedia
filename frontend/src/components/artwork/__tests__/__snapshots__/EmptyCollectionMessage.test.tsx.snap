// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCollectionMessage should render EmptyCollectionMessage component with message for logged in user 1`] = `
<div>
  <div
    class="px-4 max-w-screen-xl pt-10 pb-10 py-4 bg-white dark:bg-gray-800 shadow-md w-full rounded-lg mb-4 border border-gray-300 dark:border-gray-600 cursor-pointer text-center"
  >
    <svg
      aria-hidden="true"
      class="mx-auto w-16 h-16 mb-4 text-gray-400"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 11H5m14 0a2 2 0 012 2v6a2 2 0 01-2 2H5a2 2 0 01-2-2v-6a2 2 0 012-2m14 0V9a2 2 0 00-2-2M5 11V9a2 2 0 012-2m0 0V5a2 2 0 012-2h6a2 2 0 012 2v2M7 7h10"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <p
      class="text-xl mb-4"
    >
      Ta kolekcja jest pusta.
    </p>
    <p
      class="text-md"
    >
      <button
        class="text-blue-600 cursor-pointer bg-transparent border-0 p-0"
        type="button"
      >
        Dodawaj nowe rekordy
      </button>
       
      ręcznie lub
       
      <button
        class="text-blue-600 cursor-pointer bg-transparent border-0 p-0"
        type="button"
      >
        zaimportuj
      </button>
       
      dane z pliku Excel, aby rozpocząć organizację swojej kolekcji.
    </p>
  </div>
</div>
`;

exports[`EmptyCollectionMessage should render EmptyCollectionMessage component with message for not logged in user 1`] = `
<div>
  <div
    class="px-4 max-w-screen-xl pt-10 pb-10 py-4 bg-white dark:bg-gray-800 shadow-md w-full rounded-lg mb-4 border border-gray-300 dark:border-gray-600 cursor-pointer text-center"
  >
    <svg
      aria-hidden="true"
      class="mx-auto w-16 h-16 mb-4 text-gray-400"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 11H5m14 0a2 2 0 012 2v6a2 2 0 01-2 2H5a2 2 0 01-2-2v-6a2 2 0 012-2m14 0V9a2 2 0 00-2-2M5 11V9a2 2 0 012-2m0 0V5a2 2 0 012-2h6a2 2 0 012 2v2M7 7h10"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <p
      class="text-xl mb-4"
    >
      Ta kolekcja jest pusta.
    </p>
    <p
      class="text-md"
    >
      <span>
        <button
          class="text-blue-600 cursor-pointer bg-transparent border-0 p-0"
          type="button"
        >
          Zaloguj się
        </button>
         
        aby móc dodawać rekordy. Nie masz konta?
         
        <button
          class="text-blue-600 cursor-pointer bg-transparent border-0 p-0"
          type="button"
        >
          Zarejestruj się
        </button>
      </span>
    </p>
  </div>
</div>
`;
